x-pack/plugins/threat_intelligence/public/modules/indicators/services/fetch_indicators.ts

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  IEsSearchRequest,
+  IKibanaSearchResponse,
+  isCompleteResponse,
+  ISearchStart,
+  isErrorResponse,
+} from '@kbn/data-plugin/public';
+import { buildEsQuery, Filter, Query, TimeRange } from '@kbn/es-query';
+import { InspectorContextValue } from '../../../containers/inspector';
+import { RawIndicatorFieldId } from '../../../../common/types/indicator';
+import { THREAT_QUERY_BASE } from '../../../../common/constants';
+import { threatIndicatorNamesOriginScript, threatIndicatorNamesScript } from '../lib/display_name';
+
+export interface RawIndicatorsResponse {
+  hits: {
+    hits: any[];
+    total: number;
+  };
+}
+
+export interface Pagination {
+  pageSize: number;
+  pageIndex: number;
+  pageSizeOptions: number[];
+}
+
+export const fetchIndicators =
+  ({
+    searchService,
+    inspectorAdapters,
+  }: {
+    searchService: ISearchStart;
+    inspectorAdapters: InspectorContextValue;
+  }) =>
+  async ({
+    signal,
+    queryKey: [_key, { pagination, selectedPatterns, filters, timeRange, sorting, filterQuery }],
+  }: {
+    signal?: AbortSignal;
+    queryKey: [
+      string,
+      {
+        pagination: Pagination;
+        selectedPatterns: string[];
+        sorting: any[];
+        filters: Filter[];
+        timeRange?: TimeRange;
+        filterQuery: Query;
+      }
+    ];
```

Comment:
for readability it might be easier have separate interfaces for these 2

Reply:
things are a bit simpler now:)